src/Type/Php/SubstrDynamicReturnTypeExtension.php

```diff
@@ -37,74 +37,70 @@ public function getTypeFromFunctionCall(
+			$maybeOneLength = !(new ConstantIntegerType(1))->isSuperTypeOf($length)->no();
+		}
 
-			$constantStrings = $string->getConstantStrings();
-			if (
-				count($constantStrings) > 0
-				&& $offset instanceof ConstantIntegerType
-				&& ($length === null || $length instanceof ConstantIntegerType)
-			) {
-				$results = [];
-				foreach ($constantStrings as $constantString) {
-					if ($length !== null) {
-						if ($functionReflection->getName() === 'mb_substr') {
-							$substr = mb_substr($constantString->getValue(), $offset->getValue(), $length->getValue());
-						} else {
-							$substr = substr($constantString->getValue(), $offset->getValue(), $length->getValue());
-						}
+		$constantStrings = $string->getConstantStrings();
+		if (
+			count($constantStrings) > 0
+			&& $offset instanceof ConstantIntegerType
+			&& ($length === null || $length instanceof ConstantIntegerType)
+		) {
+			$results = [];
+			foreach ($constantStrings as $constantString) {
+				if ($length !== null) {
+					if ($functionReflection->getName() === 'mb_substr') {
+						$substr = mb_substr($constantString->getValue(), $offset->getValue(), $length->getValue());
 					} else {
-						if ($functionReflection->getName() === 'mb_substr') {
-							$substr = mb_substr($constantString->getValue(), $offset->getValue());
-						} else {
-							$substr = substr($constantString->getValue(), $offset->getValue());
-						}
+						$substr = substr($constantString->getValue(), $offset->getValue(), $length->getValue());
 					}
-
-					if (is_bool($substr)) {
-						$results[] = new ConstantBooleanType($substr);
+				} else {
+					if ($functionReflection->getName() === 'mb_substr') {
+						$substr = mb_substr($constantString->getValue(), $offset->getValue());
 					} else {
-						$results[] = new ConstantStringType($substr);
+						$substr = substr($constantString->getValue(), $offset->getValue());
 					}
 				}
 
-				return TypeCombinator::union(...$results);
+				if (is_bool($substr)) {
+					$results[] = new ConstantBooleanType($substr);
+				} else {
+					$results[] = new ConstantStringType($substr);
+				}
 			}
 
-			if ($string->isNonEmptyString()->yes() && ($negativeOffset || $zeroOffset && $positiveLength)) {
-				if ($string->isNonFalsyString()->yes() && !$maybeOneLength) {
-					return new IntersectionType([
-						new StringType(),
-						new AccessoryNonFalsyStringType(),
-					]);
+			return TypeCombinator::union(...$results);
+		}
 
-				}
+		if ($string->isNonEmptyString()->yes() && ($negativeOffset || $zeroOffset && $positiveLength)) {
+			if ($string->isNonFalsyString()->yes() && !$maybeOneLength) {
 				return new IntersectionType([
 					new StringType(),
-					new AccessoryNonEmptyStringType(),
+					new AccessoryNonFalsyStringType(),
 				]);
+
 			}
+			return new IntersectionType([
+				new StringType(),
+				new AccessoryNonEmptyStringType(),
+			]);
 		}
-
-		return null;
```

Comment:
the missing `return null` breaks the build

Reply:
Fixed